x-pack/plugins/discover_log_explorer/public/customizations/custom_dataset_selector.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { DiscoverStateContainer } from '@kbn/discover-plugin/public';
+import { DatasetSelectionHandler, DatasetSelector } from '../components/dataset_selector';
+import { DatasetsProvider, useDatasetsContext } from '../hooks/use_datasets';
+import { InternalStateProvider, useDataView } from '../hooks/use_data_view';
+import { IntegrationsProvider, useIntegrationsContext } from '../hooks/use_integrations';
+import { IDatasetsClient } from '../services/datasets';
+
+interface CustomDatasetSelectorProps {
+  stateContainer: DiscoverStateContainer;
+}
+
+export const CustomDatasetSelector = withProviders(({ stateContainer }) => {
+  // Container component, here goes all the state management and custom logic usage to keep the DatasetSelector presentational.
+  const dataView = useDataView();
+
+  const {
+    error: integrationsError,
+    integrations,
+    isLoading: isLoadingIntegrations,
+    loadMore,
+    reloadIntegrations,
+    searchIntegrations,
+    searchIntegrationsStreams,
+    sortIntegrations,
+    sortIntegrationsStreams,
+  } = useIntegrationsContext();
+
+  const {
+    datasets,
+    error: datasetsError,
+    isLoading: isLoadingStreams,
+    loadDatasets,
+    reloadDatasets,
+    searchDatasets,
+    sortDatasets,
+  } = useDatasetsContext();
+
+  /**
+   * TODO: this action will be abstracted into a method of a class adapter in a follow-up PR
+   * since we'll need to handle more actions from the stateContainer
+   */
+  const handleStreamSelection: DatasetSelectionHandler = (dataset) => {
```

Comment:
I wonder if there would be a clean way to also set the default columns to timestamp and message. 🤔 How difficult would you expect that to be?

Reply:
It should be difficult. My idea, as mentioned in the TODO, was to create an adapter layer that can create all the handlers that will need to use the stateContainer, so it should be quick to do it ideally. Shall we add this to this same implementation work?

Reply:
As with the other remaining tasks, let's try to get this PR over the line and quickly follow up with such improvements.